pkg/core/consensus/generation/generator.go

```diff
@@ -47,6 +52,8 @@ func (g *Generator) ID() uint32 {
 }
 
 // Collect `Restart` events and triggers a Generation event
-func (g *Generator) Collect(ev consensus.Event) error {
-	return g.signer.SendInternally(topics.Generation, emptyHash[:], emptyPayload, g.ID())
+func (g *Generator) Collect(_ consensus.InternalPacket) error {
+	packet := g.signer.Compose(restartFactory)
+	msg := message.New(topics.Restart, packet)
+	return g.signer.SendInternally(topics.Generation, msg, g.ID())
```

Comment:
Due to the topic being `Generation`, but the message category being `Restart`, this causes a loop. The message will end back up in the same place, causing infinite dissemination of a `Restart` message. Message category should be set to `Generation` for as long as #272 is not fixed.